src/java/org/apache/cassandra/index/sai/disk/v5/V5VectorPostingsWriter.java

```diff
@@ -101,12 +105,50 @@ public V5VectorPostingsWriter(RemappedPostings remappedPostings)
         this.remappedPostings = remappedPostings;
     }
 
-    public V5VectorPostingsWriter(Structure structure, int graphSize, Map<VectorFloat<?>, VectorPostings.CompactionVectorPostings> postingsMap)
+    /**
+     * This method describes the mapping done during construction of the graph so that we can easily create
+     * an appropriate V5VectorPostingsWriter.  No ordinal remapping is performed because (V5) compaction writes
+     * vectors to disk as they are added to the graph, so there is no opportunity to reorder the way there is
+     * in a Memtable index.
+     */
+    public static RemappedPostings describeForCompaction(Structure structure, int graphSize, Map<VectorFloat<?>, VectorPostings.CompactionVectorPostings> postingsMap)
     {
         if (structure == Structure.ONE_TO_ONE)
-            remappedPostings = new RemappedPostings(Structure.ONE_TO_ONE, graphSize - 1, graphSize - 1, null, null);
-        else
-            remappedPostings = remapPostings(postingsMap);
+        {
+            return new RemappedPostings(Structure.ONE_TO_ONE,
+                                        graphSize - 1,
+                                        graphSize - 1,
+                                        null,
+                                        null,
+                                        new OrdinalMapper.IdentityMapper(graphSize - 1));
+        }
+
+        if (structure == Structure.ONE_TO_MANY)
+        {
+            var maxOldOrdinal = postingsMap.values().stream().mapToInt(VectorPostings::getOrdinal).max().orElseThrow();
+            int maxRow = postingsMap.values().stream().flatMap(p -> p.getRowIds().stream()).mapToInt(i -> i).max().orElseThrow();
+            var extraOrdinals = new Int2IntHashMap(Integer.MIN_VALUE);
+            for (var entry : postingsMap.entrySet())
+            {
+                if (entry.getValue().getRowIds().size() == 1)
+                    continue;
+                var a = entry.getValue().getRowIds().toIntArray();
+                Arrays.sort(a);
+                int ordinal = entry.getValue().getOrdinal();
+                for (int i = 1; i < a.length; i++)
```

Comment:
Pretty minor, but I think it's worth noting here that when we construct the graph, the ordinal for a new vector is assigned the row id for that vector, and as a consequence of building in order, we sort the array and skip the first value.

Alternatively, we could add an assert that `ordinal == a[0]`.

I mention it because when I saw above that "there is no opportunity to reorder", I wasn't quite sure how this worked, so I had to double check.

Reply:
Or, we could skip sorting the array and skip the row id that equals `ordinal` in the for loop.